go-controller/pkg/ovn/kubevirt.go

```diff
@@ -56,18 +64,300 @@ func (oc *DefaultNetworkController) deleteDHCPOptions(pod *kapi.Pod) error {
+			IPPrefix:   podAddress,
+			Nexthop:    podAddress,
+			Policy:     &nbdb.LogicalRouterStaticRoutePolicyDstIP,
+			OutputPort: &outputPort,
+			ExternalIDs: map[string]string{
+				"namespace":      pod.Namespace,
+				kubevirt.VMLabel: pod.Labels[kubevirt.VMLabel],
+			},
+		}
+		if err := libovsdbops.CreateOrReplaceLogicalRouterStaticRouteWithPredicate(oc.nbClient, types.OVNClusterRouter, &ingressRoute, func(item *nbdb.LogicalRouterStaticRoute) bool {
+			matches := item.IPPrefix == ingressRoute.IPPrefix && item.Nexthop == ingressRoute.Nexthop && item.Policy != nil && *item.Policy == *ingressRoute.Policy
+			return matches
+		}); err != nil {
+			return fmt.Errorf("failed adding static route: %v", err)
+		}
+	}
+	return nil
+}
+
+func (oc *DefaultNetworkController) ensureARPProxyContainsSubnet(switchName string, subnet *net.IPNet) error {
+	lspName := ovntypes.SwitchToRouterPrefix + switchName
+	lsp, err := libovsdbops.GetLogicalSwitchPort(oc.nbClient, &nbdb.LogicalSwitchPort{Name: lspName})
+	if err != nil {
+		return err
+	}
+	arpProxy, ok := lsp.Options["arp_proxy"]
+	if !ok {
+		return fmt.Errorf("missing arp_proxy at logical switch port '%s'", lspName)
+	}
+	subnetString := subnet.String()
+	for _, arpProxyEntry := range strings.Split(arpProxy, " ") {
+		if arpProxyEntry == subnetString {
+			return nil
+		}
+	}
+	arpProxy += " " + subnetString
+	lsp.Options["arp_proxy"] = arpProxy
+	if err := libovsdbops.UpdateLogicalSwitchPortSetOptions(oc.nbClient, lsp); err != nil {
+		return fmt.Errorf("failed update logical switch port to add subnet to arp_proxy: %v", err)
+	}
+	return nil
+}
+
+func (oc *DefaultNetworkController) removeARPProxySubnet(switchName string, subnet *net.IPNet) error {
+	lspName := ovntypes.SwitchToRouterPrefix + switchName
+	lsp, err := libovsdbops.GetLogicalSwitchPort(oc.nbClient, &nbdb.LogicalSwitchPort{Name: lspName})
+	if err != nil {
+		return err
+	}
+	arpProxy, ok := lsp.Options["arp_proxy"]
+	if !ok {
+		return fmt.Errorf("missing arp_proxy at logical switch port '%s'", lspName)
+	}
+	subnetString := subnet.String()
+	arpProxyProcessed := []string{}
+	for _, arpProxyEntry := range strings.Split(arpProxy, " ") {
+		if arpProxyEntry == subnetString {
+			continue
+		}
+		arpProxyProcessed = append(arpProxyProcessed, arpProxyEntry)
+	}
+	lsp.Options["arp_proxy"] = strings.Join(arpProxyProcessed, " ")
+	if err := libovsdbops.UpdateLogicalSwitchPortSetOptions(oc.nbClient, lsp); err != nil {
+		return fmt.Errorf("failed update logical switch port to remove subnet from arp_proxy: %v", err)
+	}
+	return nil
+}
+
+func (oc *DefaultNetworkController) ensureRoutingForVM(pod *kapi.Pod) error {
+	isLiveMigrationLefover, err := kubevirt.PodIsLiveMigrationLeftOver(oc.client, pod)
+	if err != nil {
+		return err
+	}
+	if util.PodWantsHostNetwork(pod) || !kubevirt.AllowPodBridgeNetworkLiveMigration(pod.Annotations) || isLiveMigrationLefover {
+		return nil
+	}
+
+	targetNode := pod.Labels[kubevirt.NodeNameLabel]
```

Comment:
what if it doesnt exist?

Reply:
Will throw and error since it's an unexpected kubevirt behaviour.